core/esmf-aspect-model-resolver/src/main/java/org/eclipse/esmf/aspectmodel/resolver/AspectModelResolver.java

```diff
@@ -51,49 +60,40 @@
 import org.eclipse.esmf.aspectmodel.versionupdate.MigratorServiceLoader;
 import org.eclipse.esmf.aspectmodel.versionupdate.migrator.BammUriRewriter;
 import org.eclipse.esmf.samm.KnownVersion;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 import com.google.common.collect.Streams;
-import io.vavr.CheckedFunction1;
+
 import io.vavr.Value;
 import io.vavr.control.Option;
 import io.vavr.control.Try;
-import org.apache.commons.io.FilenameUtils;
-import org.apache.jena.rdf.model.Model;
-import org.apache.jena.rdf.model.ModelFactory;
-import org.apache.jena.rdf.model.Property;
-import org.apache.jena.rdf.model.RDFNode;
-import org.apache.jena.rdf.model.Resource;
-import org.apache.jena.vocabulary.RDF;
-import org.apache.jena.vocabulary.XSD;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 /**
  * Provides facilities for loading an Aspect model and resolving referenced meta model elements and
- * model elements from other Aspect models.
+ * model elements from other Aspect models
  */
 public class AspectModelResolver {
    private static final Logger LOG = LoggerFactory.getLogger( AspectModelResolver.class );
+   private static final List<Pair<Predicate<URI>, Function<URI, Try<ResolutionStrategy>>>> STRATEGY_BUILDERS = new ArrayList<>();
```

Comment:
After having read more of the PR, I think the URI here is not supposed to identify the model element to resolve (i.e. its AspectModelUrn) but is already the source for the model (URL or `file:` URI)? In this case, I don't understand the setup at all; since this information is supposed to be known only to the ResolutionStrategy; the AspectModelResolver's task is only to execute the ResolutionStrategies and collect and merge their results. Where a model "comes from" should be encapsulated by the ResolutionStrategy.